Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -61,8 +61,17 @@ namespace WebCore {
 
 static FloatRect calculateDocumentMarkerBounds(const InlineIterator::TextBoxIterator&, const MarkedText&);
 
-TextBoxPainter::TextBoxPainter(const LayoutIntegration::InlineContent& inlineContent, const InlineDisplay::Box& box, const RenderStyle& style, PaintInfo& paintInfo, const LayoutPoint& paintOffset)
-    : m_textBox(InlineIterator::BoxModernPath { inlineContent, inlineContent.indexForBox(box) })
+std::optional<TextBoxPainter> TextBoxPainter::create(const LayoutIntegration::InlineContent& inlineContent, const InlineDisplay::Box& box, const RenderStyle& style, PaintInfo& paintInfo, const LayoutPoint& paintOffset)
+{
+    InlineIterator::BoxModernPath textBox { inlineContent, inlineContent.indexForBox(box) };
+    if (!textBox.hasRenderer() || !is<RenderText>(textBox.renderer()))
+        return { };
+
+    return TextBoxPainter { textBox, style, paintInfo, paintOffset };
+}
```

Comment:
Quite likely this will just push the crash somewhere else if it even does anything. I think we need more of a root cause here.